asciidoc/courses/app-go/modules/1-driver/lessons/1-installation/lesson.adoc

```diff
@@ -0,0 +1,166 @@
+:branch: main
+
+In the link:/courses/cypher-fundamentals/[Cypher Fundamentals^] course, we cover how to query Neo4j using a language called Cypher.
+To execute a Cypher statement against a Neo4j database you will use an object called a *Driver*.
+
+To connect to and query Neo4j from within a Go application, you use the link:https://neo4j.com/developer/go[Neo4j Go Driver^].
+
+
+== Installing the Driver
+
+The Neo4j Go Driver can be installed using the `go get` command.
+
+include::{repository-raw}/{branch}/README.adoc[tag=install]
+
+
+== Importing the Library
+
+The `github.com/neo4j/neo4j-go-driver` package exports a `neo4j` object which contains several helper functions for interacting with Neo4j.
+
+
+.Importing The Driver
+[source,go,indent=0,subs="attributes+"]
+----
+include::{repository-raw}/{branch}/pkg/example/example.go[tag=import]
+----
+
+== Creating a Driver Instance
+
+The most prominent helper function on the `neo4j` object is the `.NewDriverWithContext()` function, which accepts neo4j credentials and returns a new `Driver` object.
+
+.Creating a Driver Instance
+[source,go,indent=0,role=nocopy,subs="attributes+"]
+----
+include::{repository-raw}/{branch}/pkg/example/example.go[tag=driver]
+----
+
+The function expects two arguments;
+
+1. A Connection URI - for example, `neo4j://localhost:7687` or `neo4j+s://dbhash.databases.neo4j.io`
+2. An authentication _token_
+// 3. Optionally, link:https://neo4j.com/docs/api/Go-driver/current/function/index.html#static-function-driver[additional driver configuration^].
+
+
+[TIP]
+.Authentication Types
+====
+Neo4j supports four authentication methods, basic authentication (e.g. username and password or LDAP), a base64 encoded Bearer token, a Kerberos token, or a custom authentication token.
+For more information on these options, visit link:https://neo4j.com/docs/operations-manual/current/authentication-authorization/[Authentication and Authorization^].
+====
+
+
+=== Verifying Connectivity
+
+You can verify that the connection details used during driver instantiation are correct by calling the `VerifyConnectivity()` method.
+
+This function returns a Promise which throws a `Neo.ClientError.Security.Unauthorized` error if the provided credentials are incorrect.
+
+.Verify Connectivity
+[source,go,indent=0,role=nocopy,subs="attributes+"]
+----
+include::{repository-raw}/{branch}/pkg/example/example.go[tag=verify]
+----
+
+== Sessions
+
+From the Driver object, you will open one or more **sessions**.
+A session is a client-side abstraction that borrows underlying TCP connections from a _connection pool_ held by the Driver and uses them to communicate with Neo4j.
+
+Sessions should be considered lightweight and disposable and can be opened and closed at will.
+
+You should `defer` the closing of each session  object using the `PanicOnClosureError()` function, which will release any underlying TCP connections held by the session object once the surrounding context has completed.
```

Comment:
note: `PanicOnClosureError` is just a convenience function used for the example (and it's quite crude, it panics whenever there is an error)
the more generic advice would be to call `Close` on sessions (and driver), preferably with a context that does not have a deadline (since you don't want to over-constrain the execution time required to close sessions/driver